### DISCUSSION TITLE
How to get all Graphs with their OPs in the Order they are called with GraphQL

### DISCUSSION THREAD

**Discussion by PascalStehling** [dagster-io/dagster#10613] [Q&A]
Hi Dagster Team,

im trying to extract all Graphs and the OPs in them from the whole Workspace. This is no Problem so far. The Problem comes, when I try to get the order of the OPs in which they run in the Graph, as I cant find any way to extract these Information.

So far i have the following Query:
```graphql
query GetGraphsAndOps {
  workspaceOrError {
    ... on Workspace {
      locationEntries {
        locationOrLoadError {
          ... on RepositoryLocation {
            repositories {
              name
              location {
                name
              }
              usedSolids {
                definition {
                  ... on CompositeSolidDefinition {
                    name
                    solids {
                      name
                    }
                  }
                }
              }
            }
          }
        }
      }
    }
  }
}

```

But how do I get the connection between the OPs/Solids in a Graph? It should be possible, as you do it in the Dagit UI.

Its also quite Confusing that you have the new (Asset, Graph) and the old Terms (Solid, Pipeline) and mix these the whole time. Also OP is missing in GraphQL. It would be great if there were some documentation which would explain all this.

Keep on the Good work
Best Pascal

**Comment by dpeng817** [comment]
Hey Pascal - in regards to the old terms - we do unfortunately still use the old terms in certain parts of the codebase, highly agreed it's confusing. We're in the process of switching over the internals, so hopefully should not be an issue forever.

Will follow up regarding actually getting dag ordering

**Reply by dpeng817** [reply]
You can retrieve dag ordering using the `pipelineSnapshotOrError` query:
```
pipelineSnapshotOrError(
      snapshotId: $snapshotId
      activePipelineSelector: $snapshotPipelineSelector
    ) {
      ... on PipelineSnapshot {
        id
        name
        solidHandles(parentHandleID: $requestScopeHandleID) {
            name
            inputs {
              definition {
                name
              }
              dependsOn {
                definition {
                  name
                  type {
                    displayName
                  }
                }
                solid {
                  name
                }
              }
            }
            outputs {
              definition {
                name
              }
              dependedBy {
                solid {
                  name
                }
                definition {
                  name
                  type {
                    displayName
                  }
                }
              }
            }
            }
        }
      }

```
the key fields here being dependsOn and dependedBy
